content/en/blog/_posts/2025-05-09-start-sidecar-first/index.md

```diff
@@ -0,0 +1,225 @@
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myapp
+  labels:
+    app: myapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+    spec:
+      containers:
+        - name: myapp
+          image: alpine:latest
+          command: ["sh", "-c", "sleep 3600"]
+      initContainers:
+        - name: nginx
+          image: nginx:latest
+          restartPolicy: Always
+          ports:
+            - containerPort: 80
+              protocol: TCP
+          readinessProbe:
+            exec:
+              command:
+              - /bin/sh
+              - -c
+              - exit 1 # this command always fails, keeping the container "Not Ready"
+            periodSeconds: 5
+      volumes:
+        - name: data
+          emptyDir: {}
+```
+
+The result is:
+
+```bash
+controlplane $ kubectl get pods -w
+NAME                    READY   STATUS    RESTARTS   AGE
+myapp-db5474f45-htgw5   1/2     Running   0          9m28s
+
+controlplane $ kubectl describe pod myapp-db5474f45-htgw5 
+Name:             myapp-db5474f45-htgw5
+Namespace:        default
+(...)
+Events:
+  Type     Reason     Age               From               Message
+  ----     ------     ----              ----               -------
+  Normal   Scheduled  17s               default-scheduler  Successfully assigned default/myapp-db5474f45-htgw5 to node01
+  Normal   Pulling    16s               kubelet            Pulling image "nginx:latest"
+  Normal   Pulled     16s               kubelet            Successfully pulled image "nginx:latest" in 163ms (163ms including waiting). Image size: 72080558 bytes.
+  Normal   Created    16s               kubelet            Created container nginx
+  Normal   Started    16s               kubelet            Started container nginx
+  Normal   Pulling    15s               kubelet            Pulling image "alpine:latest"
+  Normal   Pulled     15s               kubelet            Successfully pulled image "alpine:latest" in 159ms (160ms including waiting). Image size: 3652536 bytes.
+  Normal   Created    15s               kubelet            Created container myapp
+  Normal   Started    15s               kubelet            Started container myapp
+  Warning  Unhealthy  1s (x6 over 15s)  kubelet            Readiness probe failed:
+```
+
+From the above findings it’s evident that only one container is ready - and I know it can’t be the sidecar, because I’ve defined it so it’ll never be ready. I also saw that myapp has been started before the sidecar is ready. That was not the result I wanted to achieve; in this case, the main app container has a hard dependency on its sidecar.
```

Comment:
> From the above findings it’s evident that only one container is ready - and I know it can’t be the sidecar, because I’ve defined it so it’ll never be ready.

Consider rewording for clarity. Also, are there any other commands, events, etc., available that indicate which container is failing the probe?